### ISSUE TITLE
Series un-isolated when toggling local time

### ISSUE THREAD

**Issue by LeviHarrison** [prometheus/prometheus#10970]
### What did you do?

I clicked on an individual series below the graph in the graph view, isolating that series, and then toggled the "Use local time" feature

### What did you expect to see?

The series continue to be isolated on the graph like before:

![image](https://user-images.githubusercontent.com/54278938/177073003-73dcfc0c-1bf4-4082-800b-66f29101295a.png)

### What did you see instead? Under which circumstances?

The series was no longer isolated on the graph, although on the list below it showed it was still selected:

![image](https://user-images.githubusercontent.com/54278938/177072953-98cc3bfa-a4d8-49e1-abf8-64cc48ee7be4.png)

### Prometheus version

```text
Tested on v2.27 and v2.36 in Microsoft Edge/Chromium
```

**Comment by rgroothuijsen** [comment]
Checked this on the latest version, and noticed that there was a "show exemplars" button which has similar "remove filters when toggled" behavior. For the sake of consistency: should filters be maintained after toggling for both, neither, or should they have different behavior?

**Comment by LeviHarrison** [comment]
The filters should be maintained on both, so that needs to be fixed too. Thanks for pointing it out.

**Comment by beorn7** [comment]
Hello from the bug scrub.

Is this still relevant for the new UI? /cc @juliusv .

**Comment by juliusv** [comment]
Yes, still relevant for the new UI. I also haven't reimplemented some of the series selection persistence that already happens in the old UI (at least, not yet, not sure how much of it is worth the complexity).

**Comment by donaldraph** [comment]
Hey, happy to take this one if it's still open.

It looks like the isolation state lives in the uPlot instance itself, not in React or Redux, so any time the chart fully rebuilds (local time, theme, resize, a refresh) it just resets. Local time is only the most obvious trigger. Is anyone already on this? And do you have a preference on the fix, keeping it minimal so the selection survives rebuilds, or moving it into panel/URL state so it persists properly? Fine either way, whatever you'd rather.

**Comment by juliusv** [comment]
@donaldraph Good point about it happening any time the chart rebuilds, that's really not good!

> And do you have a preference on the fix, keeping it minimal so the selection survives rebuilds, or moving it into panel/URL state so it persists properly? Fine either way, whatever you'd rather.

No strong opinion. Keeping it relatively localized inside the uPlot code sounds good, unless that gets too messy and it's easier/cleaner to control it from just around that. We definitely don't need to lift this state up very high though (it doesn't need to survive page refreshes or URL shares and the like).

**Comment by donaldraph** [comment]
I opened #19532 for this. I went with the localized approach in the uplot code like you suggested, and I added a test for the visibility merge. Would appreciate a look whenever you get the chance, thanks!
